File: src/problems/75/s0.py
from typing import List


class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        counts = [0, 0, 0]
        for num in nums:
            counts[num] += 1

        r = counts[0]
        g = counts[1]
        b = counts[2]

        for i in range(r):
            nums[i] = 0

        end = r + g
        for i in range(r, end):
            nums[i] = 1

        for i in range(end, end + b):
            nums[i] = 2


data = [
    ([2, 0, 2, 1, 1, 0], [0, 0, 1, 1, 2, 2]),
    ([2, 0, 1], [0, 1, 2]),
]

for input, output in data:
    Solution().sortColors(input)
    print(input, output)

File: src/problems/75/s0.rs
impl Solution {
    pub fn sort_colors(nums: &mut Vec<i32>) {
        let mut counts = [0, 0, 0];
        for num in nums.iter_mut() {
            counts[*num as usize] += 1;
        }

        let [r, g, b] = counts;
        for i in 0..counts[0] {
            nums[i] = 0;
        }

        let end = r + g;
        for i in r..r + g {
            nums[i] = 1;
        }

        let end2 = end + b;
        for i in end..end2 {
            nums[i] = 2;
        }
    }
}
